### PR TITLE
Fix pcap dependency, str.strip() now takes an argument, add cc.get_return_value()

### PR DESCRIPTION
**Pull request by nirbheek**
commit c569ea7c97a2cfcf49f7d3c121a39e519b2726f6

str.strip() can now take a positional argument All the specified characters in the specified argument will be stripped. If unspecified, the old behaviour is used. Includes tests.

commit 6f87f94f5bbe25c913c94a4b23ba18feb143abf4

Add a new compiler method: get_return_value()

This method accepts a single function that takes no arguments and returns a single value which can be a value that can be cast to a 64-bit signed integer, or a string, and returns that value.
    
Mostly useful for running foolib_version() functions that return the currently-available version of libraries.

commit 0ae03d86b7bc836fc35d0fc692937bdb9cbdabe0

dependencies: Add version detection to pcap

--

The `PcapDependency()` changes are untested on macOS (will test that tomorrow), but we could probably ship it as-is anyway since this PR only improves things overall.

### CODE REVIEW

File: mesonbuild/dependencies/misc.py [hunk 1]
```diff
@@ -614,7 +614,9 @@ def __init__(self, environment, kwargs):
                     self.is_found = True
                     self.compile_args = fwdep.get_compile_args()
                     self.link_args = fwdep.get_link_args()
-                    self.version = '2'  # FIXME
+                    # FIXME: Test on macOS
+                    #self.version = self.get_pcap_lib_version()
```
Comment (ignatenkobrain) [inline mesonbuild/dependencies/misc.py]: __[flake8]__
* [E265] block comment should start with '# '

Comment (ignatenkobrain) [inline mesonbuild/dependencies/misc.py]: __[flake8]__
* [E265] block comment should start with '# '

File: mesonbuild/interpreter.py [hunk 2]
```diff
@@ -874,7 +875,28 @@ def get_define_method(self, args, kwargs):
         extra_args = self.determine_args(kwargs)
         deps = self.determine_dependencies(kwargs)
         value = self.compiler.get_define(element, prefix, self.environment, extra_args, deps)
-        mlog.log('Checking for value of define "%s": %s' % (element, value))
+        mlog.log('Fetching value of define "%s": %s' % (element, value))
+        return value
+
+    def get_return_value_method(self, args, kwargs):
```
Comment (jpakkane) [inline mesonbuild/interpreter.py]: This is a poor name and very confusing. "Return value" already has a specific and unambiguous meaning: the integer that the executable returns from main.

File: run_project_tests.py [hunk 3]
```diff
@@ -579,7 +579,7 @@ def check_file(fname):
     with open(fname, 'rb') as f:
         lines = f.readlines()
     for line in lines:
-        if b'\t' in line:
+        if line.startswith(b'\t'):
```
Comment (jpakkane) [inline run_project_tests.py]: Why this change? It breaks if you have an indent of `space-tab`.

File: mesonbuild/interpreter.py [hunk 4]
```diff
@@ -874,7 +875,28 @@ def get_define_method(self, args, kwargs):
...
+                "are 'string' and 'int'"
+            raise InterpreterException(m.format(rtype))
+        prefix = kwargs.get('prefix', '')
+        if not isinstance(prefix, str):
+            raise InterpreterException('Prefix argument of get_return_value() must be a string')
+        extra_args = self.determine_args(kwargs)
+        deps = self.determine_dependencies(kwargs)
+        value = self.compiler.get_return_value(fname, rtype, prefix, self.environment,
+                                               extra_args, deps)
+        mlog.log('Fetching return value of {!r}(): {!r}'.format(fname, value))
```
Comment (jpakkane) [inline mesonbuild/interpreter.py]: If this method is to be added, it must get corresponding documentation.

File: run_project_tests.py (hunk 3 above)
Comment (nirbheek) [inline run_project_tests.py]: Without this, you can't even use `\t` inside a string for testing. I can change it to a regex, but I wasn't sure if it was worth the trouble; people who mix spaces and tabs usually use `<tab><tab><spaces>`; I've never seen `<spaces><tab>`.

File: mesonbuild/interpreter.py (hunk 4 above)
Comment (nirbheek) [inline mesonbuild/interpreter.py]: Yes, I just wanted to get some feedback about the whole thing first. This method is very useful internally, but I am not sure how useful it is as an external API.

File: mesonbuild/interpreter.py (hunk 2 above)
Comment (nirbheek) [inline mesonbuild/interpreter.py]: What else would you call the value returned by a function though? Also, the return value from main in an executable is called an "exit code" or "exit status", isn't it? See: https://www.gnu.org/software/libc/manual/html_node/Exit-Status.html

**Comment by jpakkane** [comment]
⚔️ conflicts.

File: mesonbuild/dependencies/misc.py [hunk 5]
```diff
@@ -614,7 +615,9 @@ def __init__(self, environment, kwargs):
...
```
Comment (ignatenkobrain) [inline mesonbuild/dependencies/misc.py]: __[flake8]__
* [E265] block comment should start with '# '

**Comment by nirbheek** [comment]
Rebased. Do you think we should expose `get_return_value()` as external API or not? We might even want to keep it around as an undocumented function for the automated tests, and then document it or change it if a need arises.

**Comment by jpakkane** [comment]
If there is no immediate need (as in, an actual user), let's keep it private.

**Comment by nirbheek** [comment]
Updated.
